tests/generation/test_utils.py

```diff
@@ -1497,6 +1497,54 @@ def test_generate_with_head_masking(self):
                 attn_weights = out[attn_name] if attn_name == attention_names[0] else out[attn_name][-1]
                 self.assertEqual(sum([w.sum().item() for w in attn_weights]), 0.0)
 
+    # TODO (joao): this test is actually not slow :) However, it is not passing in some models (e.g. GPTNeoX) and the
+    # fix for some models is quite lengthy. Being slow means it doesn't block our push CI while we fix it.
+    @slow
+    def test_left_padding_compatibility(self):
+        # The check done in this test is fairly difficult -- depending on the model architecture, passing the right
+        # position index for the position embeddings can still result in a different output, due to numerical masking.
+        # On the other hand, for some types of position embeddings, an incorrect position index can have a minimal
+        # impact on the output.
+        # There are two tricks employed to check whether left-padding compatibility is in place:
+        # 1 - To reduce the negative impact of the numerical attention mask on a correct position index, we set the
+        # padding size to 1.
+        # 2 - To reduce the chance of false positives (i.e. passing when it should be failing), we run the check
+        # multiple times with random inputs, and it has to pass with all of them.
+        # NOTE: because of 2), there is some chance of false positives in this test.
+
+        for model_class in self.all_generative_model_classes:
+            config, _, _, _ = self._get_input_ids_and_config()
+            if config.is_encoder_decoder:
+                continue  # skip for encoder-decoder models -- they don't need left-padding compatibility
+            model = model_class(config).to(torch_device).eval()
+            signature = inspect.signature(model.forward).parameters.keys()
+
+            no_failures = True
+            for _ in range(30):
```

Comment:
do you know how long this test takes in general (with `30` in the loop).

I would prefer not to use `30` - if `5` sounds good to you.

Reply:
That doesn't answer the question of how long it takes ;-) It's okay if we detect a false positive with the test being flaky compared to having a very long test.

Reply:
@sgugger @ydshieh 
Ah, on my local machine it takes 13.51s to run this test on ALL models. So less than 1 second per decoder-only model :)

Reply:
Sound good, but maybe let me try on our CI runner which is not as fast as your local machine :-)

Reply:
Can we still put a smaller number before merging? Like 10? The tests are executed very often so we will see flaky failures. There are also tools to repeat a given tests multiple times in pytest.